source_code/src/OLEDMINI/bitstreammini.c

```diff
@@ -45,38 +45,36 @@ void miniBistreamInit(bitstream_mini_t* bs, uint8_t height, uint16_t width, uint
     bs->width = width;
     bs->height = height;
     bs->dataCounter = 0;
-    bs->bufferInd = sizeof(bs->buffer);
-    bs->dataSize = (uint16_t)width * (((uint16_t)height+7) >> BITSTREAM_PIXELS_PER_BYTE_BITSHIFT);
+    bs->dataSize = (uint16_t)width * (((uint16_t)height+7) / BITSTREAM_PIXELS_PER_BYTE);
 }
 
 /*! \fn     bsGetNextByte(bitstream_mini_t* bs)
  *  \brief  Return the next data byte from flash
  *  \param  bs      pointer to initialized bitstream context to get the next word from
  *  \return next data word, or 0 if end of data reached
  */
-uint8_t miniBistreamGetNextByte(bitstream_mini_t* bs)
-{
-    // Are you requesting bytes when you've already read everything?
-    if (bs->dataCounter < bs->dataSize) 
-    {
-        bs->dataCounter++;
-        
-        // Check if we need to fetch new data from the SPI flash
-        if(bs->bufferInd >= sizeof(bs->buffer))
-        {
-            // Fetch new data from external flash
-            flashRawRead(bs->buffer, bs->addr, sizeof(bs->buffer));
-            bs->addr += sizeof(bs->buffer);
-            bs->bufferInd = 0;
-            //usbPrintf_P(PSTR("bistream buffer: %02x %02x %02x %02x %02x %02x"), bs->buffer[0], bs->buffer[1], bs->buffer[2], bs->buffer[3], bs->buffer[4], bs->buffer[5]);
-        }
-        
-         return bs->buffer[bs->bufferInd++];
-    }
-    else
-    {
-        return 0;
-    }
-}
+ uint8_t miniBistreamGetNextByte(bitstream_mini_t* bs)
+ {
+     // Are you requesting bytes when you've already read everything?
+     if (bs->dataCounter < bs->dataSize)
+     {
+         uint16_t buffer_index = bs->dataCounter % sizeof(bs->buffer);
```

Comment:
> It will if BITSTREAM_BUFFER_SIZE isn't a power of 2... and are you actually 100% the compiler optimizes it?
As in every microcontroller code it is known that buffer sizes are meant to be power of 2. And in out code it is and there is not a single reason why we would change it to something not equal to 2.

> Have you tried it?
Sure it was a thing of 10 seconds to check that. I always do that, as for the other function I have NOT used % as it gives an overhead (but this has other reasons).

> no, everytime we need to display a picture that has hundreds of pixels.
As I said if you read a single byte always without a buffer at all you can only see it for the main menu, as it updates fast. A normal fullscreen picture is still not a problem. And the main menu is still reasonable fast for the fact that in this case the whole spi flash read process is executed for every byte

> why is that? do you actually have measurements to back your claims?
For the same reasons as above. You cannot see it. I've slowed down the SPI speed down to KHz instead of MHz and then you could see it. But that spi optimization is really not required on 4MHz.

> Saying that it's fine because your general feeling is that it isn't slowing down is not the way to go.
I can see your doubts, but I dont get the real problem. A (forced, tested) slowdown doesnt break the application. So the only thing that matters is, if you see it. And if you don't you are good to go.